regulation-worker/internal/delete/api/api.go

```diff
@@ -76,6 +106,17 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destConfig map
 		return model.JobStatusFailed
 	}
 
+	if isOAuthEnabled {
+		isOAuthTokenExpired := isTokenExpired(jobResp)
+		if isOAuthTokenExpired {
+			err = api.refreshOAuthToken(destination.Name, job.WorkspaceID, oAuthDetail)
```

Comment:
We return an error. Yes, I think it will be re-tried after sometime. @saurav-malani can you provide insight here ?

Reply:
What will happen if we have only 1 job and its retry backoff becomes higher than the token's validity period? Won't we enter an endless loop?

Reply:
yes @atzoum, this is a possible known issue with the current implementation. In fact, I did raise this concern is our discussion with the transformation team when you and Leo where also in the call.

Reply:
is there something we are afraid of to not retry the request a second time?

Reply:
no, I think given that in response from transformer we get status corresponding to each user in the batch. It should be quite simple. And, I don't see any problem.

Reply:
Currently the userIds for which the failure happens is not being sent in the response. But it can be captured & server needs to make changes to it's response schema to capture that detail & process accordingly which I think is not in scope of this PR.

As far as retry goes, would 1 retry solve the problem or should we retry multiple times ?!